schemas/+previousimaging/Segmentation.m

```diff
@@ -0,0 +1,1697 @@
+        
+        roi_data.roi_global_xy              = globalXY(:,iROI);
+        roi_data.roi_is_in_chunks           = [];   
+        roi_data.roi_spatial                = [];
+        
+        trace_data.time_constants           = data.cnmf.timeConstants{iROI};
+        trace_data.init_concentration       = data.cnmf.initConcentration{iROI};
+        trace_data.dff_roi                  = nan(1,totalFrames);
+        trace_data.dff_surround             = nan(1,totalFrames);
+        trace_data.spiking                  = nan(1,totalFrames);
+        trace_data.dff_roi_is_significant   = nan(1,totalFrames);
+        trace_data.dff_roi_is_baseline      = nan(1,totalFrames);
+        
+        % now look in file chunks and fill activity etc
+        for iChunk = 1:numel(chunkdata)
+          % find roi in chunks
+          localIdx                          = data.chunk.globalID == iROI;
+          if sum(localIdx) == 0; continue; end
+          roi_data.roi_is_in_chunks         = [roi_data.roi_is_in_chunks iChunk];
+            
+          % activity traces
+          frameIdx                                    = chunkRange(iChunk,1):chunkRange(iChunk,2);
+          uniqueData                                  = chunkdata{iChunk}.cnmf.uniqueData(localIdx,:);
+          uniqueBase                                  = halfSampleMode(uniqueData');
+          surroundData                                = chunkdata{iChunk}.cnmf.surroundData(localIdx,:);
+          trace_data.dff_roi(frameIdx)                = uniqueData / uniqueBase - 1;
+          trace_data.dff_surround(frameIdx)           = surroundData / uniqueBase - 1;
+          trace_data.spiking(frameIdx)                = chunkdata{iChunk}.cnmf.spiking(localIdx,:);
+          trace_data.dff_roi_is_significant(frameIdx) = chunkdata{iChunk}.cnmf.isSignificant(localIdx,:);
+          trace_data.dff_roi_is_baseline(frameIdx)    = chunkdata{iChunk}.cnmf.isBaseline(localIdx,:);
+          
+          % roi: shape and morphological classification
+          if isempty(roi_data.roi_spatial)
+            roi_data.roi_spatial      = reshape(full(chunkdata{iChunk}.cnmf.spatial(:,localIdx)),chunkdata{iChunk}.cnmf.region.ImageSize);
+            roi_data.surround_spatial = reshape(full(chunkdata{iChunk}.cnmf.surround(:,localIdx)),chunkdata{iChunk}.cnmf.region.ImageSize);
+            morpho_data.morphology    = char(chunkdata{iChunk}.cnmf.morphology(localIdx));
+          end
+        end
+        
+        % insert in tables
+        insert(previousimaging.SegmentationRoi, roi_data)
+        insert(previousimaging.SegmentationRoiMorphologyAuto, morpho_data)
+        insert(previousimaging.Trace, trace_data)
+      end
+
+    end
+  end
+end
+
+%% ------------------------------------------------------------------------
+%% file chunk selection
+%% ------------------------------------------------------------------------
+
+function fileChunk = selectFileChunks(key,chunk_cfg)
```

Comment:
This script is a bit too long. I think what would be helpful is to move the cnmf processing functions into a separate package inside imaging. For example, +imaging/+cnmf and each of the function could be a separate file. It would be awesome if Sue-Ann could help provide some documentation or we schedule a meeting with her to write down what each function does.

Reply:
All right, I agree of something like +imaging/+cnmf would be great. 
I will check with Sue Ann when I get to this point in the "unified" pipeline